src/sentry/profiles/consumer.py

```diff
@@ -0,0 +1,143 @@
+from collections import defaultdict
+from typing import Any, Dict, MutableMapping, Optional, Sequence
+
+import msgpack
+from confluent_kafka import Producer
+from django.conf import settings
+
+from sentry.lang.native.processing import process_payload
+from sentry.utils import json, kafka_config
+from sentry.utils.batching_kafka_consumer import AbstractBatchWorker, BatchingKafkaConsumer
+from sentry.utils.kafka import create_batching_kafka_consumer
+
+
+def get_profiles_consumer(
+    topic: Optional[str] = None, **options: Dict[str, str]
+) -> BatchingKafkaConsumer:
+    config = settings.KAFKA_TOPICS[settings.KAFKA_PROFILES]
+    producer = Producer(
+        kafka_config.get_kafka_producer_cluster_options(config["cluster"]),
+    )
+    return create_batching_kafka_consumer(
+        {topic},
```

Comment:
`topic` is marked `Optional`, but `create_batching_kafka_consumer` doesn't look like it can handle `None`